rfd/0229-agent-inventory-ui.md

```diff
@@ -0,0 +1,233 @@
+      "id": "2",
+      "type": "agent",
+      "agent": {
+        "name": "kube1",
+        "version": "16.5.4",
+        "services": ["kube"],
+        "upgrader": {
+          "type": "kube",
+          "version": "v2",
+          "group": "group1"
+        }
+      }
+    },
+    {
+      "id": "3",
+      "type": "bot",
+      "bot": {
+        "name": "bot1",
+        "version": "16.1.3",
+        "service": "kubernetes",
+        "health": "Unknown"
+      }
+    },
+    {
+      "id": "4",
+      "type": "agent",
+      "agent": {
+        "name": "agent1",
+        "version": "16.1.3",
+        "services": ["kube"],
+        "upgrader": {
+          "type": "kube",
+          "version": "v2",
+          "group": "group2"
+        }
+      }
+    }
+  ],
+  "updaterGroups": ["group1", "group2"],
+  "totalCount": 4,
+  "startKey": ""
+}
+```
+
+#### RPC Service and Proto messages
+
+```protobuf
+service InventoryService {
+  // ListUnifiedInstances returns a page of Agents and Bot Instances. This API will refuse any requests when the cache is unhealthy or not yet
+  // fully initialized.
+  rpc ListUnifiedInstances(ListUnifiedInstancesRequest) returns (ListUnifiedInstancesResponse);
+}
+
+// ListUnifiedInstancesRequest is the request for listing instances.
+message ListUnifiedInstancesRequest {
+  // page_size is the size of the page to return.
+  int32 page_size = 1;
+  // page_token is the next_page_token value returned from a previous ListUnifiedInstances request, if any.
+  string page_token = 2;
+  // filters specify optional search criteria to limit which instances should be returned.
+  ListUnifiedInstancesFilter filter = 3;
+}
+
+// ListUnifiedInstancesResponse is the response from listing instances.
+message ListUnifiedInstancesResponse {
+  // items is the list of instances (agents or bot instances) returned.
+  repeated UnifiedInstanceItem items = 1;
+  // next_page_token contains the next page token to use as the start key for the next page of instances.
+  string next_page_token = 2;
+}
+
+// ListUnifiedInstancesFilter provides a mechanism to refine ListUnifiedInstances results.
+message ListUnifiedInstancesFilter {
+  // search is a basic string search query which will filter results by name (hostname for agents, bot name for bot instances).
+  string search = 1;
+  // advanced_search is an advanced search query using predicate language.
+  string advanced_search = 2;
+  // kinds is the kinds of instances to return. If ommitted, both agent instances and bot instances will be returned.
+  repeated string kinds = 3;
+  // services is the list of services to filter agents by. Only applicable to agent instances.
+  repeated string services = 4;
+  // updater_groups is the list of updater groups to filter instances by. Only applicable to agent instances.
+  repeated string updater_groups = 5;
+  // upgraders is the list of upgraders to filter instances by. Only applicable to agent instances.
+  repeated string upgraders = 6;
```

Comment:
What measures will we put in place such that if/when tbot support managed updated, i.e. https://github.com/gravitational/teleport/pull/60478/files, that these apply to bots as well?

Reply:
There don't need to be any special measures put in place now, the service logic will just need to be updated to also filter bot instances by it if/when managed updates are added for them. I'll remove the "only applicable to..." part of the comment.